search/search_service/proxy/es_proxy_v2_1.py

```diff
@@ -303,3 +313,152 @@ def _build_should_query(self, resource: Resource, query_term: str) -> List[Q]:
             rank_feature_queries.append(rank_feature_query)
 
         return rank_feature_queries
+
+    def _search_highlight(self,
+                          resource: Resource,
+                          search: Search,
+                          highlight_options: Dict[Resource, HighlightOptions]) -> Search:
+        # get highlighting options for resource
+        highlighting_enabled = highlight_options.get(resource).enable_highlight
+
+        if highlighting_enabled and resource != Resource.USER:
+            # default highlighted fields
+            search = search.highlight('name',
+                                      type=DEFAULT_HIGHLIGHTER,
+                                      number_of_fragments=0)
+            search = search.highlight('description',
+                                      type=DEFAULT_HIGHLIGHTER,
+                                      number_of_fragments=5, order='none')
+            if resource == Resource.TABLE:
+                search = search.highlight('columns.general',
+                                          type=DEFAULT_HIGHLIGHTER,
+                                          number_of_fragments=5, order='score')
+            if resource == Resource.DASHBOARD:
+                search = search.highlight('chart_names',
+                                          type=DEFAULT_HIGHLIGHTER,
+                                          number_of_fragments=5, order='score')
+                search = search.highlight('query_names',
+                                          type=DEFAULT_HIGHLIGHTER,
+                                          number_of_fragments=5, order='score')
+
+        return search
+
+    def _format_response(self, page_index: int,
```

Comment:
do something to DRY this up. My recommendation is a "utility function" that is reused by both proxies, where the older V2 proxy will simply not have any highlighting in its response.